pandas/tseries/resample.py

```diff
@@ -323,6 +323,11 @@ def aggregate(self, arg, *args, **kwargs):
                                                *args,
                                                **kwargs)
 
+        # if arg was a string, _aggregate called resampler's _downsample or
+        # _groupby_and_agg methods, which would've already applied the loffset
+        if not isinstance(arg, compat.string_types):
```

Comment:
unclear why this is happening here, and not in ``.downsample`` (and ``.upsample``) instead. checking the arg type is not very robust (and does't follow a good pattern).